### PR TITLE
Added equals and hash code methods for ServiceName and tests

### PR DESCRIPTION
**Pull request by SG-O**
Added equals and hashCode methods to ServiceName and added new tests for these.
This should resolve https://github.com/eclipse-vertx/vertx-grpc/issues/58

### CODE REVIEW

File: vertx-grpc-common/src/main/java/io/vertx/grpc/common/impl/ServiceNameImpl.java [hunk 1]
```diff
@@ -61,4 +63,17 @@ public String pathOf(String method) {
       return '/' + packageName + '.' + name + '/' + method;
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
```
Comment (vietj) [inline vertx-grpc-common/src/main/java/io/vertx/grpc/common/impl/ServiceNameImpl.java:68]: can you add curly brace in statements ?

Comment (SG-O) [reply vertx-grpc-common/src/main/java/io/vertx/grpc/common/impl/ServiceNameImpl.java:68]: I hope the changes I made follow the code-style you desire.

**Comment by vietj** [comment]
can you backport the fix for 4.x branch @SG-O ?

**Comment by SG-O** [comment]
Sure thing! I'll do it in another PR.

**Pull request merged**